src/common/ScrollReveal.tsx

```diff
@@ -0,0 +1,188 @@
+// Credits: https://www.reactbits.dev/text-animations/scroll-reveal
+
+import { gsap } from 'gsap'
+import { ScrollTrigger } from 'gsap/ScrollTrigger'
+import React, { ReactNode, RefObject, useEffect, useMemo, useRef } from 'react'
+
+gsap.registerPlugin(ScrollTrigger)
+
+interface ScrollRevealProps {
+  children: ReactNode
+  scrollContainerRef?: RefObject<HTMLElement>
+  enableBlur?: boolean
+  baseOpacity?: number
+  baseRotation?: number
+  blurStrength?: number
+  containerClassName?: string
+  textClassName?: string
+  rotationEnd?: string
+  wordAnimationEnd?: string
+}
+
+const ScrollReveal: React.FC<ScrollRevealProps> = ({
+  children,
+  scrollContainerRef,
+  enableBlur = true,
+  baseOpacity = 0.1,
+  baseRotation = 3,
+  blurStrength = 4,
+  containerClassName = '',
+  textClassName = '',
+  rotationEnd = 'bottom bottom',
+  wordAnimationEnd = 'bottom bottom',
+}) => {
+  const containerRef = useRef<HTMLHeadingElement>(null)
+
+  // Allow highlighting portions of text
+  // and applying a className in the format [some text][classNameWithoutDot].
+  // See example in ProjectSection.
+  const splitText = useMemo(() => {
+    const text = typeof children === 'string' ? children : ''
+
+    // Define a regular expression to match patterns like [text][className]
+    const regex = /\[([^\]]*)\]\[([^\]]*)\]/g
+
+    const parts = []
+    let lastIndex = 0
+
+    let match
+    while ((match = regex.exec(text)) !== null) {
+      if (match.index > lastIndex) {
+        const preText = text.slice(lastIndex, match.index)
+
+        // Split the text into words and spaces
+        const wordsWithSpaces = preText.split(/(\s+)/)
+        wordsWithSpaces.forEach((word, i) => {
+          if (word.trim()) {
+            parts.push(
+              <span key={`${word}-${i}`} className='word inline-block'>
+                {word}
+              </span>,
+            )
+            return
+          }
+          // If it's just a space, push the space as it is
+          parts.push(word)
+        })
+      }
+
+      // process the matched pattern [text][className]
+      parts.push(
+        <span key={match[0]} className={match[2] + ' word'}>
+          {match[1]}
+        </span>,
```

Comment:
maybe destructure array to give elements proper names?
e.g. `const [fullMatch, className, word] = match`